ui/src/datatypes.ts

```diff
@@ -0,0 +1,81 @@
+  { name: 'decimal', uri: expand('xsd:decimal'), params: defaultParams },
+  { name: 'integer', uri: expand('xsd:integer'), params: defaultParams },
+  { name: 'long', uri: expand('xsd:long'), params: defaultParams },
+  { name: 'int', uri: expand('xsd:int'), params: defaultParams },
+  { name: 'short', uri: expand('xsd:short'), params: defaultParams },
+  { name: 'byte', uri: expand('xsd:byte'), params: defaultParams },
+  { name: 'nonNegativeInteger', uri: expand('xsd:nonNegativeInteger'), params: defaultParams },
+  { name: 'positiveInteger', uri: expand('xsd:positiveInteger'), params: defaultParams },
+  { name: 'unsignedLong', uri: expand('xsd:unsignedLong'), params: defaultParams },
+  { name: 'unsignedInt', uri: expand('xsd:unsignedInt'), params: defaultParams },
+  { name: 'unsignedShort', uri: expand('xsd:unsignedShort'), params: defaultParams },
+  { name: 'unsignedByte', uri: expand('xsd:unsignedByte'), params: defaultParams },
+  { name: 'nonPositiveInteger', uri: expand('xsd:nonPositiveInteger'), params: defaultParams },
+  { name: 'negativeInteger', uri: expand('xsd:negativeInteger'), params: defaultParams },
+  { name: 'number', uri: expand('xsd:double'), params: defaultParams },
+  { name: 'duration', uri: expand('xsd:duration'), params: [...defaultParams, 'format'] },
+  { name: 'dayTimeDuration', uri: expand('xsd:dayTimeDuration'), params: [...defaultParams, 'format'] },
+  { name: 'yearMonthDuration', uri: expand('xsd:yearhMonthDuration'), params: [...defaultParams, 'format'] },
+  { name: 'float', uri: expand('xsd:float'), params: defaultParams },
+  { name: 'gDay', uri: expand('xsd:gDay'), params: [...defaultParams, 'format'] },
+  { name: 'gMonth', uri: expand('xsd:gMonth'), params: [...defaultParams, 'format'] },
+  { name: 'gMonthDay', uri: expand('xsd:gMonthDay'), params: [...defaultParams, 'format'] },
+  { name: 'gYear', uri: expand('xsd:gYear'), params: [...defaultParams, 'format'] },
+  { name: 'gYearMonth', uri: expand('xsd:gYearMonth'), params: [...defaultParams, 'format'] },
+  { name: 'hexBinary', uri: expand('xsd:hexBinary'), params: defaultParams },
+  { name: 'QName', uri: expand('xsd:QName'), params: defaultParams },
+  { name: 'string', uri: expand('xsd:string'), params: ['language', ...defaultParams] },
+  { name: 'normalizedString', uri: expand('xsd:normalizedString'), params: defaultParams },
+  { name: 'token', uri: expand('xsd:token'), params: defaultParams },
+  { name: 'language', uri: expand('xsd:language'), params: defaultParams },
+  { name: 'Name', uri: expand('xsd:Name'), params: defaultParams },
+  { name: 'NMTOKEN', uri: expand('xsd:NMTOKEN'), params: defaultParams },
+  { name: 'xml', uri: expand('rdf:xml'), params: defaultParams },
+  { name: 'html', uri: expand('rdf:html'), params: defaultParams },
+  { name: 'json', uri: expand('csvw:json'), params: defaultParams },
+  { name: 'time', uri: expand('xsd:time'), params: [...defaultParams, 'format'] }
+]
```

Comment:
😲

Reply:
We'll probably want to cherry-pick the types we want to show to the user. As far as I know, some of these will never be used, and others are even redundant (int vs integer)...

Reply:
Does seems a little too csvw-centric how what we've been trying to be agnostic. Fine for now but it's just the observation

Reply:
Yeah I agree, but it seemed like the pragmatic thing to do. I think?